compiler/rustc_mir/src/transform/instrument_coverage.rs

```diff
@@ -54,69 +50,131 @@ pub(crate) fn provide(providers: &mut Providers) {
 /// are still included in the total `num_counters` or `num_expressions`.) Simply counting the
 /// calls may not work; but computing the number of counters or expressions by adding `1` to the
 /// highest ID (for a given instrumented function) is valid.
+///
+/// This visitor runs twice, first with `add_missing_operands` set to `false`, to find the maximum
+/// counter ID and maximum expression ID based on their enum variant `id` fields; then, as a
+/// safeguard, with `add_missing_operands` set to `true`, to find any other counter or expression
+/// IDs referenced by expression operands, if not already seen.
+///
+/// Ideally, every expression operand in the MIR will have a corresponding Counter or Expression,
```

Comment:
```suggestion
/// Ideally, every counter expression id operand in the MIR will have a corresponding Counter or Expression,
```

Reply:
Point taken, but I reworded it a little differently. Let me know if it's better or not.